src/librustc_resolve/lib.rs

```diff
@@ -407,6 +410,17 @@ fn reduce_impl_span_to_impl_keyword(cm: &CodeMap, impl_span: Span) -> Span {
     impl_span
 }
 
+fn generate_fn_name_span(cm: &CodeMap, span: Span) -> Option<Span> {
+    let prev_span = cm.span_extend_to_prev_str(span, "fn", true);
+    cm.span_to_snippet(prev_span).map(|snippet| {
+        let len: usize = snippet.chars()
```

Comment:
This could be `snippet.find(|c| !c.is_alphanumeric() && *c != '_').expect("no label after `fn`")`